src/traffic_server/traffic_server.cc

```diff
@@ -509,32 +521,38 @@ init_memory_tracker(const char *config_var, RecDataT /* type ATS_UNUSED */, RecD
 static void
 proxy_signal_handler(int signo, siginfo_t *info, void *ctx)
 {
-  if ((unsigned)signo < countof(signal_received)) {
-    signal_received[signo] = true;
-  }
+  Debug("SIGNAL", "got signal on thread %lx", pthread_self());
+  if (term_by_signal_threads[pthread_self()]) {
```

Comment:
I'm actually a little worried using `[]` operator  here, since it will create a new key value pair if the key doesn't exist in the map, but `contains` is only available in c++20, so I'm thinking maybe I should use `find`?

Reply:
```
if (auto spot = term_by_signal_threads.find(pthread_self()) ; spot != term_by_signal_threads.end()) {
   ...
}
```
?